WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt

```diff
@@ -559,11 +552,46 @@ class PostListViewModel @Inject constructor(
 
         return PostAdapterItem(
                 data = postData,
-                onSelected = { handlePostButton(PostListButton.BUTTON_EDIT, post) },
-                onButtonClicked = { handlePostButton(it, post) }
+                onSelected = {
+                    trackAction(PostListButton.BUTTON_EDIT, post, AnalyticsTracker.Stat.POST_LIST_ITEM_SELECTED)
+                    handlePostButton(PostListButton.BUTTON_EDIT, post)
+                },
+                onButtonClicked = {
+                    trackAction(it, post, AnalyticsTracker.Stat.POST_LIST_BUTTON_PRESSED)
+                    handlePostButton(it, post)
+                }
         )
     }
 
+    private fun trackAction(buttonType: Int, postData: PostModel, statsEvent: Stat) {
+        val properties = HashMap<String, Any?>()
+        if (!postData.isLocalDraft) {
+            properties["post_id"] = postData.remotePostId
+        }
+
+        when (buttonType) {
+            PostListButton.BUTTON_EDIT -> {
+                properties["action"] = "edit"
+                properties[AnalyticsUtils.HAS_GUTENBERG_BLOCKS_KEY] = PostUtils
+                        .contentContainsGutenbergBlocks(postData.content)
+            }
+            PostListButton.BUTTON_RETRY -> properties["action"] = "retry"
```

Comment:
Should we get all these `properties["action"]` assignments out of the `when` and assign it to the result of `when` instead?

We might even consider changing these buttons to an enum as part of post filters project. That'd let us get rid of the `else` statement and also make sure we are handling every possible case.

Reply:
Yep, the properties["action"] should definitely not be copy-pasted 10 times:D... I'll fix it right away.

Yes I also think it might be worth updating to enum. I just didn't want to change it here since there was still a chance it'd go to 10.8.

Reply:
> Yes I also think it might be worth updating to enum. I just didn't want to change it here since there was still a chance it'd go to 10.8.

Fully agreed with the decision of not doing this as part of this PR.